pkg/kubelet/dockershim/docker_stats_windows.go

```diff
@@ -0,0 +1,119 @@
+// +build windows
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockershim
+
+import (
+	dockertypes "github.com/docker/docker/api/types"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+	"time"
+)
+
+// ContainerStats returns stats for a container stats request based on container id.
+func (ds *dockerService) ContainerStats(containerStatsRequest *runtimeapi.ContainerStatsRequest) (*runtimeapi.ContainerStatsResponse, error) {
+	containerStats, err := ds.createContainerStats(containerStatsRequest.ContainerId)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &runtimeapi.ContainerStatsResponse{
+		Stats: containerStats,
+	}, nil
+}
+
+// ListContainerStats returns stats for a list container stats request based on a filter.
+func (ds *dockerService) ListContainerStats(containerStatsRequest *runtimeapi.ListContainerStatsRequest) (*runtimeapi.ListContainerStatsResponse, error) {
+	filter := &runtimeapi.ContainerFilter{}
+
+	if containerStatsRequest.Filter != nil {
+		filter.Id = containerStatsRequest.Filter.Id
+		filter.PodSandboxId = containerStatsRequest.Filter.PodSandboxId
+		filter.LabelSelector = containerStatsRequest.Filter.LabelSelector
+	}
+
+	containers, err := ds.ListContainers(filter)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var stats []*runtimeapi.ContainerStats
+	for _, container := range containers {
+		containerStats, err := ds.createContainerStats(container.Id)
+
+		if err != nil {
+			return nil, err
+		}
+
+		stats = append(stats, containerStats)
+	}
+
+	return &runtimeapi.ListContainerStatsResponse{
+		Stats: stats,
+	}, nil
+}
+
+func (ds *dockerService) createContainerStats(containerID string) (*runtimeapi.ContainerStats, error) {
```

Comment:
I think `getContainerStats` is closer to what this function does.

Reply:
Fixed, renamed it to `getContainerStats`